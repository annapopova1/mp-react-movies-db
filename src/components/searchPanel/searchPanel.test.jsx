import React from 'react';
import SearchPanel from './searchPanel';

describe('<SearchPanel/>', () => {
  test('should render SearchPanel component', () => {
    const handler = jest.fn();
    const searchPanel = shallow(<SearchPanel
      searchByParam="director"
      searchString="test str"
      searchHandler={handler}
    />);
    expect(searchPanel).toMatchSnapshot();
  });

  test('should handle search actions', () => {
    const param = 'director';
    const searchStr = 'test str';
    const handler = jest.fn();

    const searchPanel = mount(<SearchPanel
      searchByParam={param}
      searchString={searchStr}
      searchHandler={handler}
    />);

    const btn = searchPanel.find('button').first().simulate('click');
    expect(handler).toBeCalledWith({
      searchByParam: param,
      searchString: searchStr,
    });

    handler.mockClear();

    // search by title with another text
    searchPanel.find('.nav-link').first().simulate('click');
    const input = searchPanel.find('#searchBox').first().simulate('keyUp', {
      keyCode: 80,
    });
    input.getDOMNode().value = 'changed text';
    input.simulate('blur');
    btn.simulate('click');
    expect(handler).toBeCalledWith({
      searchByParam: 'title',
      searchString: 'changed text',
    });

    handler.mockClear();

    // search by clicking enter
    input.getDOMNode().value = 'new text';
    input.simulate('blur');
    input.simulate('keyUp', {
      keyCode: 13,
    });
    expect(handler).toBeCalledWith({
      searchByParam: 'title',
      searchString: 'new text',
    });
  });
});
