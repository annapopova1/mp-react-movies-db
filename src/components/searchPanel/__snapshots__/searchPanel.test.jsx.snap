// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPanel/> should render SearchPanel component 1`] = `
<section>
  <h5>
    FIND YOUR MOVIE
  </h5>
  <div
    className="form-label-group"
  >
    <input
      className="form-control"
      id="searchBox"
      onBlur={[Function]}
      onKeyUp={[Function]}
      placeholder="Search..."
      type="text"
    />
  </div>
  <NavPanel
    direction="left"
    links={
      Array [
        Object {
          "active": false,
          "handler": [Function],
          "param": "title",
          "title": "TITLE",
        },
        Object {
          "active": true,
          "handler": [Function],
          "param": "director",
          "title": "DIRECTOR",
        },
      ]
    }
    navBtn={
      Object {
        "handler": [Function],
        "title": "SEARCH",
      }
    }
    primaryBrand={null}
    secondaryBrand="SEARCH BY"
  />
</section>
`;
